src/auth/auth-context.ts

```diff
@@ -41,6 +39,15 @@ export const setTokens = (
   }
 };
 
+export const removeTokens = (): void => {
+  try {
+    window.localStorage.removeItem(tokenStorageKey);
+  } catch (error) {
```

Comment:
I wonder if we would wish to not catch this error here? Or catch it but throw another error? Because the user of removeTokens function would probably like to know if the operation succeeded or not?

Reply:
🤔 Should I actually detect if the user even has invalid tokens in the store? Then we would not need the try-catch-block

Reply:
Like this maybe: 
```suggestion
  const saveAuthTokes = (tokens: OptionalAuthTokens) => {
    storeTokens(tokens);
    setAuthTokens(tokens);
    setLoading(false);
  };

  const onError = (e) => {
    // eslint-disable-next-line no-console
    console.error(`Authentication failed: ${e.message}`);
    setAuthTokens(undefined);
    setLoading(false);
  };

  useEffect(() => {
    const queryParams: ParsedUrlQuery = querystring.parse(
      window.location.search.replace('?', '')
    );
    const authTokensFromQuery: OptionalAuthTokens = pickAuthParams(queryParams);
    const storedAuthTokens: OptionalAuthTokens = getTokens();

    if (storedAuthTokens) {
      removeTokens();
    }

    if (authTokensFromQuery) {
      api
        .testAuthCredentials(authTokensFromQuery)
        .then(() => saveAuthTokes(authTokensFromQuery))
        .catch(onError);
    } else if (storedAuthTokens) {
      api
        .testAuthCredentials(storedAuthTokens)
        .then(() => saveAuthTokes(storedAuthTokens))
        .catch(onError);
    } else {
      setAuthTokens(undefined);
      setLoading(false);
    }
  }, []);

```

Reply:
That could work. I just wonder if in theory it would be possible for something to remove the item after successfull check but before removal? So in theory it could always throw :thinking:

Reply:
Yes, I will change that the removeTokens will always throw